### PR TITLE
Apple Silicon support

### PR DESCRIPTION
**Pull request by wyhaya**
Issue: #928 

* Use `shaderc` on `aarch64-apple-darwin` target
* Update `winit`: https://github.com/rust-windowing/winit/commit/3a077ff21125e9b130ab18596e9a363b5d9dcb86

~~`bevy_audio` is currently unavailable, May be related to https://github.com/RustAudio/coreaudio-sys/pull/45~~

### CODE REVIEW

**Comment by joshuajbouw** [comment]
Sweet, yeah that confirms exactly why results that I had when trying to compile for M1.
* There -can- be Audio support, you MUST compile from the same architecture though, so that would limit it to only being able to build it on a M1 chip from what I understand?
* `shaderc` for M1 makes sense from what I gathered too, and it does compile.

Maybe waiting for Winit to do an update would be a good idea first though. At the very least, I would suggest having the Cargo.toml git patch put in as a conditional much like WASM's.

**Comment by wyhaya** [comment]
Currently `bevy_audio` fails to build on the M1 chip.
```
error: failed to run custom build command for `coreaudio-sys v0.2.7`

Caused by:
  process didn't exit successfully: `/Users/../bevy/target/debug/build/coreaudio-sys../build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=COREAUDIO_SDK_PATH
  cargo:rustc-link-lib=framework=AudioToolbox
  cargo:rustc-link-lib=framework=CoreAudio
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS

  --- stderr
  thread 'main' panicked at 'libclang error; possible causes include:
  - Invalid flag syntax
  - Unrecognized flags
  - Invalid flag arguments
  - File I/O errors
  - Host vs. target architecture mismatch
  If you encounter an error missing from this list, please file an issue or a PR!', /Users/../bindgen-0.56.0/src/ir/context.rs:531:15
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```
Related Issues: https://github.com/rust-lang/rust-bindgen/issues/1211

**Comment by joshuajbouw** [comment]
I see, ok good to know. Hopefully they fix it soon.

**Comment by wyhaya** [comment]
To build `bevy_audio` on the M1 chip, need to wait for an upstream update

```
rodio
    cpal
        coreaudio-rs
            coreaudio-sys
```

**Comment by MichaelHills** [comment]
Yep soon https://github.com/RustAudio/coreaudio-sys/pull/46

**Comment by MichaelHills** [comment]
`coreaudio-sys` has been published, is audio good to go now?

File: crates/bevy_render/src/shader/shader.rs [hunk 1]
```diff
@@ -42,7 +50,7 @@ impl Into<shaderc::ShaderKind> for ShaderStage {
     }
 }
 
-#[cfg(target_os = "ios")]
+#[cfg(any(target_os = "ios", all(target_os = "macos", target_arch = "aarch64")))]
```
Comment (frewsxcv) [inline crates/bevy_render/src/shader/shader.rs:53]: The function below was updated recently to return a `Result`. You can copy what I added here to make this work on Apple Silicon:

https://github.com/bevyengine/bevy/pull/1027

I created my pull request before I saw yours, oops

**Comment by wyhaya** [comment]
@MichaelHills bevy_audio now works fine!

**Comment by wyhaya** [comment]
Close this PR, looks like #1027 is working better

**Pull request closed without merging**